src/corefx/System.Globalization.Native/pal_collation.c

```diff
@@ -587,6 +591,142 @@ int32_t GlobalizationNative_IndexOfOrdinalIgnoreCase(
     return result;
 }
 
+/*
+ collation element is an int used for sorting. It consists of 3 components:
+    * primary - first 16 bits, representing the base letter
+    * secondary - next 8 bits, typically an accent
+    * tertiary - last 8 bits, typically the case
+
+An example (the numbers are made up to keep it simple)
+    a: 1 0 0
+    ą: 1 1 0
+    A: 1 0 1
+    Ą: 1 1 1
+
+    this method returns a mask that allows for characters comparison using specified Collator Strength
+*/
+static int32_t GetCollationElementMask(UColAttributeValue strength)
+{
+    assert(strength >= UCOL_SECONDARY);
+
+    switch (strength)
+    {
+        case UCOL_PRIMARY:
+            return UCOL_PRIMARYORDERMASK;
+        case UCOL_SECONDARY:
+            return UCOL_PRIMARYORDERMASK | UCOL_SECONDARYORDERMASK;
+        default:
+            return UCOL_PRIMARYORDERMASK | UCOL_SECONDARYORDERMASK | UCOL_TERTIARYORDERMASK;
+    }
+}
+
+static int32_t SimpleStartsWith_Iterators(UCollationElements* pPatternIterator, UCollationElements* pSourceIterator, UColAttributeValue strength)
```

Comment:
Could these functions return 1-byte `bool` type defined in C99 `stdbool.h` (already included on top of this file)?

Reply:
See https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-guidelines.md#unix-shims

> Stick to data types which are guaranteed not to vary in size across flavors.

All of our existing shims are using `int32_t` for booleans, which guarantees the size will be consistent.